x-pack/plugins/security_solution/public/common/components/cell_actions/index.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CellActions, useDataGridColumnsCellActions } from '@kbn/cell-actions';
+import type {
+  CellActionsProps,
+  UseDataGridColumnsCellActions,
+  UseDataGridColumnsCellActionsProps,
+} from '@kbn/cell-actions';
+import React, { useMemo } from 'react';
+import type { CellActionFieldValue, CellActionsData } from '@kbn/cell-actions/src/types';
+import type { SecurityMetadata } from '../../../actions/types';
+import { SecurityCellActionsTrigger, SecurityCellActionType } from '../../../actions/constants';
+import { SourcererScopeName } from '../../store/sourcerer/model';
+import { useGetFieldSpec } from '../../hooks/use_get_field_spec';
+
+// bridge exports for convenience
+export * from '@kbn/cell-actions';
+export { SecurityCellActionsTrigger, SecurityCellActionType };
+
+export interface SecurityCellActionsData {
+  /**
+   * The field name
+   */
+  field: string;
+
+  value: CellActionFieldValue;
+}
+
+export interface SecurityCellActionsProps
+  extends Omit<CellActionsProps, 'data' | 'metadata' | 'disabledActionTypes' | 'triggerId'> {
+  scopeId?: SourcererScopeName;
+  data: SecurityCellActionsData | SecurityCellActionsData[];
+
+  triggerId: SecurityCellActionsTrigger; // can not use SecurityCellActionsTrigger, React.FC Validation throws error for some reason
+  disabledActionTypes?: SecurityCellActionType[];
+  metadata?: SecurityMetadata;
+}
+
+export interface UseDataGridColumnsSecurityCellActionsProps
+  extends UseDataGridColumnsCellActionsProps {
+  triggerId: SecurityCellActionsTrigger;
+  disabledActionTypes?: SecurityCellActionType[];
+  metadata?: SecurityMetadata;
+}
+
+export const useDataGridColumnsSecurityCellActions: UseDataGridColumnsCellActions<UseDataGridColumnsSecurityCellActionsProps> =
+  useDataGridColumnsCellActions;
+
+export const SecurityCellActions: React.FC<SecurityCellActionsProps> = ({
+  scopeId = SourcererScopeName.default,
+  data,
+  ...props
+}) => {
+  const getFieldSpec = useGetFieldSpec(scopeId);
+  // Make a dependency key to prevent unnecessary re-renders when data object is defined inline
+  // It is necessary because the data object is an array or object and useMemo would always re-render
+  const dependencyKey = Array.isArray(data) ? data : `${data.field} - ${data.value}`;
```

Comment:
For the array case, we will re-render every time. Also, `field` is always a string here, but the `value` can also be an array, this interpolation will omit changes in the value. 
Why don't we use just `JSON.stringify(data)` directly? This structure is relatively small.

Reply:
Good point. I was considering that the array would be memoized. But that might require too much boilerplate and be inconsistent with how it works for a single value. `JSON.stringify(data)` would be the best key, but I wonder if it could be slower than the extra rerenders we are trying to prevent. Isn't there a more performative option?
It would be nice to have a `useDeepEqualMemo`for this scenario.

What do you think about this alternative?
```ts
const buildDependencyKey = (data: SecurityCellActionsData[]) =>
  data
    .map(({ field, value }) => `${field}:${Array.isArray(value) ? value.join('_') : value}`)
    .join(',');
```

Reply:
I implemented your suggestioin.